TinyOAuth1/TinyOAuth.cs

```diff
@@ -342,40 +333,35 @@ public async Task<RequestTokenInfo> GetRequestTokenAsync()
 				//	oauth_token_secret:
 				//The Token Secret.
 
-				string oauthToken = null;
-				string oauthTokenSecret = null;
 				//string oauthAuthorizeUrl = null;
 
-				var keyValPairs = responseText.Split('&');
-
-				for (var i = 0; i < keyValPairs.Length; i++)
-				{
-					var splits = keyValPairs[i].Split('=');
-					switch (splits[0])
-					{
-						case "oauth_token":
-							oauthToken = splits[1];
-							break;
-						case "oauth_token_secret":
-							oauthTokenSecret = splits[1];
-							break;
-							// TODO: Handle this one?
-							//case "xoauth_request_auth_url":
-							//	oauthAuthorizeUrl = splits[1];
-							//	break;
-					}
-				}
+				var keyValPairs = HttpUtility.ParseQueryString(responseText);
 
+			    var oauthToken = keyValPairs["oauth_token"];
+			    var oauthTokenSecret = keyValPairs["oauth_token_secret"];
+                
 				return new RequestTokenInfo
 				{
 					RequestToken = oauthToken,
-					RequestTokenSecret = oauthTokenSecret
+					RequestTokenSecret = oauthTokenSecret,
+                    AdditionalParams = ToDictionary(keyValPairs)
 				};
 			}
+
 			throw new Exception("Empty response text when getting the request token");
 		}
 
-		private static string ConcatList(IEnumerable<string> source, string separator)
+	    private static IDictionary<string, string> ToDictionary(NameValueCollection col)
```

Comment:
Would this be better as an extension method in its own file?

Reply:
Sure.